resources/plugincontent.py

```diff
@@ -111,20 +112,36 @@ def play_track(self):
                     "title":track['name'],
                     "genre": track["genre"],
                     "year": track["year"],
-                    "tracknumber": track["track_number"],
                     "album": track['album']["name"],
                     "artist": track["artist"],
                     "rating": track["rating"],
                     "duration": track["duration_ms"]/1000
                 }
+
+        if include_track_number:
+            infolabels["tracknumber"] = track["track_number"]
+
         li.setInfo( type="Music", infoLabels=infolabels)
         li.setProperty("spotifytrackid",track['id'])
         if KODI_VERSION > 15:
             li.setContentLookup(False)
         li.setProperty('do_not_analyze', 'true')
         
+        return url, li
+
+    def play_track(self):
+        track = self.sp.track(self.trackid, market=self.usercountry)
+        _url, li = self.parse_spotify_track(track)
         xbmcplugin.setResolvedUrl(handle=self.addon_handle, succeeded=True, listitem=li)
-        
+
+    def play_track_radio(self):
+        player = SpotifyRadioPlayer()
+        player.set_parent(self)
+        player.set_seed_track_id(self.trackid)
+        player.play()
+        while(not xbmc.abortRequested):
```

Comment:
xbmc.abortRequested is deprecated a while ago. The new way is to instantiate the xbmc.Monitor class to monitor the shutdown. That said, wouldn't it be better to hand over the playback to the background service instead of blocking the plugin call ?